src/tilt-estimator.cpp

```diff
@@ -0,0 +1,51 @@
+#include <state-observation/observer/tilt-estimator.hpp>
+
+namespace stateObservation
+{
+
+  TiltEstimator::TiltEstimator(double alpha, double beta, double gamma)
+    : ZeroDelayObserver(9, 6), alpha_(alpha), beta_(beta), gamma_(gamma),
+      p_S_C(Vector3::Zero()), R_S_C(Matrix3::Identity()), v_S_C(Vector3::Zero()), w_S_C(Vector3::Zero()), v_C(Vector3::Zero())
+  {
+  }
+  
+  void TiltEstimator::setMeasurement(const Vector3 ya_k, const Vector3 yg_k, TimeIndex k)
+  {
+    ObserverBase::MeasureVector y_k(6);
+    y_k << ya_k, yg_k;
+
+    ZeroDelayObserver::setMeasurement(y_k, k);
+  }
+
+  ObserverBase::StateVector TiltEstimator::oneStepEstimation_()
+  {
+    TimeIndex k = this->x_.getTime();
+    
+    BOOST_ASSERT(this->y_.size() > 0 && this->y_.checkIndex(k+1) && "ERROR: The measurement vector is not set");
+
+    Vector3 ya = getMeasurement(k+1).head(3);
```

Comment:
Here you can use the compile-time versions of the head and tail
```suggestion
    Vector3 ya = getMeasurement(k+1).head<3>();
```